src/app/components/CreateTask.tsx

```diff
@@ -0,0 +1,287 @@
+const status = ['Todo', 'In Progress', 'In review', 'Done']
+
+//This is mock data which will be replaced after API integration.
+const statusIcons: { [key: string]: ReactNode } = {
+  Todo: <TodoIcon />,
+  'In Progress': <InprogressIcon />,
+  'In review': <InreviewIcon />,
+  Done: <DoneIcon />,
+}
+
+//This is mock data which will be replaced after API integration.
+const assignee: IAssignee[] = [
+  {
+    type: '',
+    name: 'No Assignee',
+  },
+  {
+    img: 'https://avatar.iran.liara.run/public/1',
+    type: 'Internal users',
+    name: 'Jacob Jones',
+  },
+  {
+    img: 'https://avatar.iran.liara.run/public/2',
+    type: 'Internal users',
+    name: 'Wade Warren',
+  },
+  {
+    img: 'https://avatar.iran.liara.run/public/3',
+    type: 'Clients',
+    name: 'Brroklyn Simmons',
+  },
+]
+
+export const CreateTask = () => {
+  const [displayStatus, setDisplayStatus] = useState(false)
+  const [statusValue, setStatusValue] = useState<string | null>(status[0])
+  const [inputStatusValue, setInputStatusValue] = useState('')
+
+  const setStatusRef = useFocusableInput(displayStatus)
+
+  const [displayAssignee, setDisplayAssignee] = useState(false)
+  const [assigneeValue, setAssigneeValue] = useState<IAssignee>(assignee[0])
+  const [inputAssigneeValue, setInputAssigneeValue] = useState<string>('')
+
+  const setAssigneeRef = useFocusableInput(displayAssignee)
+
+  return (
+    <Box
+      sx={{
+        margin: '0 auto',
+        background: '#ffffff',
+        position: 'absolute',
+        top: '50%',
+        left: '50%',
+        transform: 'translate(-50%, -50%)',
+        boxShadow: '0px 16px 70px 0px rgba(0, 0, 0, 0.5)',
+        border: (theme) => `1px solid ${theme.color.borders.border2}`,
+        borderRadius: '4px',
+        width: '685px',
+      }}
+    >
+      <Stack
+        direction="row"
+        alignItems="center"
+        justifyContent="end"
+        sx={{
+          border: (theme) => `1px solid ${theme.color.borders.borderDisabled}`,
+        }}
+      >
+        <AppMargin size={SizeofAppMargin.MEDIUM} ptb="12px">
+          <Close
+            sx={{ color: (theme) => theme.color.gray[500], cursor: 'pointer' }}
+            onClick={() => store.dispatch(setShowModal())}
+          />
+        </AppMargin>
+      </Stack>
+
+      <AppMargin size={SizeofAppMargin.MEDIUM} ptb="16px">
+        <Stack direction="column" rowGap={1}>
+          <Typography variant="md">Task name</Typography>
+          <StyledTextField type="text" padding="8px 0px" />
+        </Stack>
+        <Stack direction="column" rowGap={1} m="16px 0px">
+          <Typography variant="md">Description</Typography>
+          <StyledTextField
+            type="text"
+            placeholder="Add description..."
+            multiline
+            rows={6}
+            inputProps={{ style: { resize: 'vertical' } }}
+          />
+        </Stack>
+
+        <Stack direction="row" columnGap={3} position="relative">
```

Comment:
Can we further modularize this component? This might grow to be hard to maintain in the future.

Reply:
547be2f1122d88405631bc849bcd922d63f7ee30 commit addresses the comment @rrojan